src/main/php/PHPMD/PHPMD.php

```diff
@@ -152,18 +148,20 @@ public function setIgnorePattern(array $ignorePatterns)
 
 
     /**
-     * Add a list of ignore patterns that is used to exclude directories from
+     * Add a list of ignore patterns which is used to exclude directories from
      * the source analysis.
      *
      * @param array<string> $ignorePatterns List of ignore patterns.
-     * @return void
+     * @return self
+     * @since 2.9.0
      */
     public function addIgnorePatterns(array $ignorePatterns)
     {
         $this->ignorePatterns = array_merge(
             $this->ignorePatterns,
             $ignorePatterns
         );
+        return $this;
```

Comment:
I would add a blank line before statements.
```suggestion

        return $this;
```